### PR TITLE
keep track of more Gcstats

### PR DESCRIPTION
**Pull request by chflood**
Keep track of mark/sweep/TTSP (time to safepoint) times for reporting in benchmarks.

### CODE REVIEW

File: base/timing.jl [hunk 1]
```diff
@@ -16,9 +16,18 @@ struct GC_Num
...
+    ttsp            ::Int64
+    max_ttsp        ::Int64
+    sweep_time      ::Int64
+    mark_time       ::Int64
+    total_sweep_time  ::Int64
+    total_mark_time   ::Int64
 end
 
 gc_num() = ccall(:jl_gc_num, GC_Num, ())
+reset_gc_stats() = ccall(:jl_gc_reset_stats,Cvoid,())
```
Comment (vchuravy) [inline base/timing.jl:30]:
```suggestion
reset_gc_stats() = ccall(:jl_gc_reset_stats, Cvoid, ())
```

File: src/gc.c [hunk 2]
```diff
@@ -3067,6 +3074,7 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
     }
 
     // 3. walk roots
+    uint64_t start_mark_time = jl_hrtime();
```
Comment (vchuravy) [inline src/gc.c:3077]: We should be consistent and do this next to `JL_PROBE_GC_MARK_BEGIN`

File: src/gc.c [hunk 3]
```diff
@@ -3079,7 +3087,7 @@ static int _jl_gc_collect(jl_ptls_t ptls, jl_gc_collection_t collection)
     gc_num.since_sweep += gc_num.allocd;
     JL_PROBE_GC_MARK_END(scanned_bytes, perm_scanned_bytes);
```
Comment (vchuravy) [inline src/gc.c:3088]: This should be `endmark`

**Comment by oscardssmith** [comment]
Time to merge this?

File: base/timing.jl [hunk 4]
```diff
@@ -18,9 +18,16 @@ struct GC_Num
     full_sweep      ::Cint
     max_pause       ::Int64
     max_memory      ::Int64
+    ttsp            ::Int64
```
Comment (vtjnash) [inline base/timing.jl:21]: could we write out these names?

**Comment by chflood** [comment]
Is it just time_to_safepoint (ttsp) that you don't like?

File: base/timing.jl (hunk 4 above)
Comment (vtjnash) [reply base/timing.jl:21]: yes, just ttsp

**Review by oscardssmith — APPROVED** [review]

File: base/timing.jl [hunk 5]
```diff
@@ -18,9 +18,16 @@ struct GC_Num
     full_sweep      ::Cint
     max_pause       ::Int64
     max_memory      ::Int64
+    time_to_safepoint             ::Int64
+    max_time_to_safepointp        ::Int64
```
Comment (oscardssmith) [inline base/timing.jl:22]: @chflood typo?

**Comment by chflood** [comment]
Aw man, yes, sorry

**Pull request merged**